### ISSUE TITLE
From required to optional in child class 

### ISSUE THREAD

**Issue by wasya243** [typegoose/typegoose#907]
Hello, I need your help regarding this:

```
class A {
	@prop({ required: true })
	a!: string;

	@prop({ required: true })
	b!: string;

	@prop({ required: true })
	c!: string
}

// some quick way to make a, b and c optional without overriding them?
class B extends A {}
```

Is there any one liner to have B class that have the same props as A class, but optional? It's not limited to transition required -> optional, but overall do we have any means to make such transitions?

**Comment by hasezoey** [comment]
> Is there any one liner to have B class that have the same props as A class, but optional? It's not limited to transition required -> optional, but overall do we have any means to make such transitions?

TL;DR: no there is not; if you want such a thing you likely will have to make the schema from the class, loop through all paths and set your options and build the model.
to my knowledge there is no such thing in mongoose directly
see #791
